namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -0,0 +1,80 @@
+package io.buoyant.namer.dnssrv
+
+import java.net.InetSocketAddress
+
+import com.twitter.finagle._
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.logging.Logger
+import com.twitter.util.Activity.State
+import com.twitter.util._
+import org.xbill.DNS
+
+import scala.collection.concurrent.TrieMap
+
+class DnsSrvNamer(prefix: Path, resolver: DNS.Resolver, timer: Timer, refreshInterval: Duration, stats: StatsReceiver)
+  extends Namer {
+
+  private val log = Logger.get("dnssrv")
+  private val cache = TrieMap.empty[Path, Var[State[NameTree[Name]]]]
```

Comment:
Take it or leave it: consider [com.twitter.util.Memoize](https://twitter.github.io/util/docs/com/twitter/util/Memoize$.html#apply[A,B](f:A=>B):A=>B) as an alternative to `map.getOrElseUpdate`.